### PR TITLE
Hello classification, classification_async, hello_reshape_ssd python samples with API 2.0

### PR DESCRIPTION
**Pull request by mgrdv**
Hello classification, classification_async, hello_reshape_ssd python samples with API 2.0
Issues:  [CVS-73317] and [CVS-73327 ]

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]
please address issues found at pre-commit checks

File: samples/python/hello_classification/hello_classification.py [hunk 1]
```diff
@@ -34,70 +36,80 @@ def main():
...
     log.info(f'Reading the network: {args.model}')
     # (.xml and .bin files) or (.onnx file)
-    net = ie.read_network(model=args.model)
+    
+    model = core.read_model(model=args.model)
 
-    if len(net.input_info) != 1:
+    # ---------------------------Step 3. Configure image_tensor----------------------------------------------------------
+    
+    if len(model.get_parameters()) != 1:
```
Comment (eaidova) [inline samples/python/hello_classification/hello_classification.py:51]:
```suggestion
    if len(model.inputs) != 1:
```

File: samples/python/hello_classification/hello_classification.py [hunk 2]
```diff
@@ -34,70 +36,80 @@ def main():
...
 
-    # ---------------------------Step 8. Process output--------------------------------------------------------------------
+    # ---------------------------Step 7. Process output--------------------------------------------------------------------
     # Generate a label list
     if args.labels:
         with open(args.labels, 'r') as f:
             labels = [line.split(',')[0].strip() for line in f]
 
-    res = res[out_blob]
+    predictions = next(iter(np.array(res)))
```
Comment (eaidova) [inline samples/python/hello_classification/hello_classification.py:109]: now infer result is dict Node is key, data is value (np.array)
```suggestion
    predictions = next(iter(res.values))
```

File: samples/python/hello_classification/hello_classification.py [hunk 3]
```diff
@@ -34,70 +36,80 @@ def main():
...
+    # 1) Set input tensor information:
+    # - input() provides information about a single model input
+    # - precision of tensor is supposed to be 'u8'
+    # - layout of data is 'NHWC'
+    # - set static spatial dimensions to input tensor to resize from
+    preproc.input() \
+           .tensor() \
+           .set_element_type(image_tensor.element_type) \
+           .set_layout(tensor_layout) \
+           .set_spatial_static_shape(h, w)
```
Comment (vladimir-dudnik) [inline samples/python/hello_classification/hello_classification.py:78]:
```suggestion
    preproc.input().tensor() \
           .set_element_type(image_tensor.element_type) \
           .set_layout(tensor_layout) \
           .set_spatial_static_shape(h, w)
```

File: samples/python/hello_classification/hello_classification.py [hunk 4]
```diff
@@ -34,70 +36,80 @@ def main():
...
-
-    # ---------------------------Step 5. Create infer request--------------------------------------------------------------
-    # load_network() method of the IECore class with a specified number of requests (default 1) returns an ExecutableNetwork
-    # instance which stores infer requests. So you already created Infer requests in the previous step.
-
-    # ---------------------------Step 6. Prepare input---------------------------------------------------------------------
-    original_image = cv2.imread(args.input)
-    image = original_image.copy()
-    _, _, h, w = net.input_info[input_blob].input_data.shape
+    exec_net = core.compile_model(model=model, device_name=args.device)
```
Comment (vladimir-dudnik) [inline samples/python/hello_classification/hello_classification.py:96]:
```suggestion
    compiled_model = core.compile_model(model=model, device_name=args.device)
```

File: samples/python/hello_classification/hello_classification.py [hunk 5]
```diff
@@ -34,70 +36,80 @@ def main():
...
-    # Add N dimension to transform to NCHW
-    image = np.expand_dims(image, axis=0)
-
-    # ---------------------------Step 7. Do inference----------------------------------------------------------------------
+    # ---------------------------Step 6. Create infer request--------------------------------------------------------------
     log.info('Starting inference in synchronous mode')
-    res = exec_net.infer(inputs={input_blob: image})
 
-    # ---------------------------Step 8. Process output--------------------------------------------------------------------
+    res = exec_net.infer_new_request(inputs={0: image_tensor})
```
Comment (vladimir-dudnik) [inline samples/python/hello_classification/hello_classification.py:101]:
```suggestion
    res = compiled_model.infer_new_request(inputs={0: image_tensor})
```

File: samples/python/hello_classification/hello_classification.py [hunk 6]
```diff
@@ -34,70 +36,80 @@ def main():
...
+           .tensor() \
+           .set_element_type(image_tensor.element_type) \
+           .set_layout(tensor_layout) \
+           .set_spatial_static_shape(h, w)
+    # 2) Adding explicit preprocessing steps:
+    # - convert layout to 'NCHW' (from 'NHWC' specified above at tensor layout)
+    # - apply linear resize from tensor spatial dims to model spatial dims
+    preproc.input().preprocess().resize(ResizeAlgorithm.RESIZE_LINEAR)
+    # 3) Here we suppose model has 'NCHW' layout for input
+    preproc.input().network().set_layout(Layout("NCHW"))
```
Comment (vladimir-dudnik) [inline samples/python/hello_classification/hello_classification.py:84]:
```suggestion
    preproc.input().network().set_layout(Layout('NCHW'))
```

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
follow new `model` naming instead old `network` and make pre-commit checks happy

File: samples/python/hello_classification/hello_classification.py [hunk 7]
```diff
@@ -34,70 +36,79 @@ def main():
...
-    if len(net.input_info) != 1:
+    model = core.read_model(model=args.model)
+
+    # ---------------------------Step 3. Configure image_tensor----------------------------------------------------------
+
+    if len(model.inputs) != 1:
         log.error('Sample supports only single input topologies')
         return -1
-    if len(net.outputs) != 1:
+    if model.get_output_size() != 1:
```
Comment (jiwaszki) [inline samples/python/hello_classification/hello_classification.py:54]: What is wrong with `len(model.outputs())`?

File: samples/python/hello_classification/hello_classification.py [hunk 8]
```diff
@@ -8,7 +8,9 @@
 
 import cv2
 import numpy as np
-from openvino.inference_engine import IECore
+from openvino.runtime import Core, Tensor, Layout
+from openvino.runtime.impl import Type
```
Comment (jiwaszki) [inline samples/python/hello_classification/hello_classification.py:12]: `impl` has been dropped in https://github.com/openvinotoolkit/openvino/pull/9096

Just import `Type` from runtime module.

File: samples/python/classification_sample_async/classification_sample_async.py [hunk 9]
```diff
@@ -40,77 +40,94 @@ def main():
...
-    for i in range(num_of_input):
-        exec_net.requests[i].async_infer({input_blob: input_data[i]})
 
-    # ---------------------------Step 8. Process output--------------------------------------------------------------------
+    infer_queue = AsyncInferQueue(compiled_model, num_of_input)
+
+    results = []
+    
+    def completion_callback(request, data_id):
+        results.append((data_id, request.results))
```
Comment (eaidova) [inline samples/python/classification_sample_async/classification_sample_async.py:122]:
```suggestion
    results = {}
    
    def completion_callback(request, data_id):
        results[data_id] = next(iter(request.results.values()))
```

File: samples/python/classification_sample_async/classification_sample_async.py [hunk 10]
```diff
@@ -119,45 +136,29 @@ def main():
...
-                probability_indent = ' ' * (len('classid') - len(str(class_id)) + 1)
-                label_indent = ' ' * (len('probability') - 8) if args.labels else ''
-                label = labels[class_id] if args.labels else ''
-                log.info(f'{class_id}{probability_indent}{probs[class_id]:.7f}{label_indent}{label}')
-            log.info('')
-
-            output_queue.remove(i)
+    for i in output_queue:
+            
+        res = np.array(list(results[output_queue[i]][1].values()))
```
Comment (eaidova) [inline samples/python/classification_sample_async/classification_sample_async.py:141]:
```suggestion
        res = results[i]
```

Comment (eaidova) [reply samples/python/classification_sample_async/classification_sample_async.py:141]: BTW, you can use range directly in loop, without necessary to create output queue variable

**Review by dpigasin — APPROVED** [review]

**Review by vladimir-dudnik — APPROVED** [review]

File: samples/python/classification_sample_async/classification_sample_async.py [hunk 11]
```diff
@@ -2,164 +2,128 @@
...
+# --------------------------- Step 6. Create infer request queue ------------------------------------------------------
+    log.info('Starting inference in asynchronous mode')
+    infer_queue = AsyncInferQueue(compiled_model, len(input_tensors))
+    infer_queue.set_callback(completion_callback)
 
-        if len(output_queue) == 0:
-            break
+# --------------------------- Step 7. Do inference --------------------------------------------------------------------
+    for i, input_tensor in enumerate(input_tensors):
+        infer_queue.start_async({0: input_tensor}, args.input[i])
```
Comment (jiwaszki) [inline samples/python/classification_sample_async/classification_sample_async.py:123]: In fact you are able to make it even more python-friendly as `start_async(...)` is called from here:
https://github.com/openvinotoolkit/openvino/blob/eb94fe27a61f211aec453f6f0721f4f61703e263/src/bindings/python/src/openvino/runtime/ie_api.py#L84
And underlaying function is converting numpy arrays to Tensors in runtime:
https://github.com/openvinotoolkit/openvino/blob/eb94fe27a61f211aec453f6f0721f4f61703e263/src/bindings/python/src/openvino/runtime/ie_api.py#L35
So you can change this loop to something like:
```suggestion
    for i, image in enumerate(resized_images):
        infer_queue.start_async({0: np.expand_dims(image, 0)}, args.input[i])
```
However the change could have effect on performance of this loop, as all numpy actions and Tensor creations will be done here. What do you win? Removal of lines 88-89 where Tensors are being created, or even more lines if you introduce resizing in this loop as well. Keep in mind - this is only example to show how API can work - for performance we have benchmark_app.

File: samples/python/hello_classification/hello_classification.py [hunk 12]
```diff
@@ -2,121 +2,111 @@
...
 
-    # ---------------------------Step 7. Do inference----------------------------------------------------------------------
+# --------------------------- Step 5. Loading model to the device -----------------------------------------------------
+    log.info('Loading the model to the plugin')
+    compiled_model = core.compile_model(model, device_name)
+
+# --------------------------- Step 6. Create infer request and do inference synchronously -----------------------------
     log.info('Starting inference in synchronous mode')
-    res = exec_net.infer(inputs={input_blob: image})
+    output_tensor = compiled_model.infer_new_request({0: input_tensor})
```
Comment (jiwaszki) [inline samples/python/hello_classification/hello_classification.py:85]: I think name `results` is better as later on you iter over it. If you were to iter over `output_tensor` you would access specific values of it.

As well you can use raw numpy array as input for this function. That will require some refactoring above.

**Comment by jiwaszki** [comment]
Overall it is a green light for me. Just some small comments about Python API capabilities.

**Review by dpigasin — APPROVED** [review]

File: samples/python/classification_sample_async/classification_sample_async.py (hunk 11 above)
Comment (vladimir-dudnik) [reply samples/python/classification_sample_async/classification_sample_async.py:122]: done

File: samples/python/hello_classification/hello_classification.py (hunk 12 above)
Comment (vladimir-dudnik) [reply samples/python/hello_classification/hello_classification.py]: done

**Review by dpigasin — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/201

**Pull request merged**